src/dmd/dmodule.d

```diff
@@ -1422,7 +1421,7 @@ extern (C++) final class Module : Package
             memcpy(todo, deferred.tdata(), len * Dsymbol.sizeof);
             deferred.setDim(0);
 
-            for (size_t i = 0; i < len; i++)
+            foreach (i; 0..len)
```

Comment:
This loop calls dsymbolSemantic, so better leave it as a `for`.

Reply:
len is a local variable in this function, dsymbolSemantic can't possibly change that.